src/bootstrap/builder.rs

```diff
@@ -662,7 +662,7 @@ impl<'a> Builder<'a> {
             cargo.env("RUSTDOC_LIBDIR", self.rustc_libdir(self.compiler(2, self.config.build)));
         }
 
-        if mode == Mode::Tool {
+        if [Mode::ToolRustc, Mode::ToolStd, Mode::ToolTest].iter().any(|m| &mode == m) {
```

Comment:
Let's implement an `is_tool` method on `Tool`.

Reply:
By this, do you mean we should use a closure `is_tool` that checks if mode is in `ToolRustc, ToolStd, etc`?

Reply:
Not quite -- method on `Tool`, not a closure.

Reply:
It makes sense now. Thanks!